clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/consumer/AbstractKafkaConsumerConfigProperties.java

```diff
@@ -0,0 +1,162 @@
+     * set client id or alternatively the value set via {@link #setDefaultClientIdPrefix(String)}, the given
+     * consumerName and a newly created UUID.
+     *
+     * Note: This method should be called for each new consumer, ensuring that a unique client id is used.
+     *
+     * @param consumerName A name for the consumer to include in the added {@code client.id} property.
+     * @return a copy of the consumer configuration with the applied properties or an empty map if neither a consumer
+     *         client configuration was set with {@link #setConsumerConfig(Map)} nor common configuration properties
+     *         were set with {@link #setCommonClientConfig(Map)}.
+     * @throws NullPointerException if consumerName is {@code null}.
+     */
+    public final Map<String, String> getConsumerConfig(final String consumerName) {
+        Objects.requireNonNull(consumerName);
+
+        if (commonClientConfig == null && consumerConfig == null) {
+            return new HashMap<>();
+        }
+
+        final Map<String, String> newConfig = new HashMap<>();
+        if (commonClientConfig != null) {
+            newConfig.putAll(commonClientConfig);
+        }
+        if (consumerConfig != null) {
+            newConfig.putAll(consumerConfig);
+        }
+
+        adaptConfiguration(newConfig);
+
+        Optional.ofNullable(keyDeserializerClass).ifPresent(serializerClass -> overrideConfigProperty(newConfig,
+                ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, serializerClass.getName()));
+
+        Optional.ofNullable(valueDeserializerClass).ifPresent(serializerClass -> overrideConfigProperty(newConfig,
+                ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, serializerClass.getName()));
+
+        setUniqueClientId(newConfig, consumerName, ConsumerConfig.CLIENT_ID_CONFIG);
+        return newConfig;
+    }
+
+    /**
+     * Adapt the properties.
+     * <p>
+     * Subclasses may overwrite this method to set expected configuration values. The default implementation does
+     * nothing.
+     *
+     * @param config The consumer configuration to be adapted.
+     */
+    protected void adaptConfiguration(final Map<String, String> config) {
+
+    }
+
+    /**
+     * Sets the timeout for polling records.
+     * <p>
+     * The default value of this property is {@link #DEFAULT_POLL_TIMEOUT}.
+     *
+     * @param pollTimeoutMillis The maximum number of milliseconds to wait.
+     * @throws IllegalArgumentException if poll timeout is negative.
+     */
+    public final void setPollTimeout(final long pollTimeoutMillis) {
+        if (pollTimeoutMillis < 0) {
+            throw new IllegalArgumentException("poll timeout must not be negative");
+        } else {
+            this.pollTimeout = pollTimeoutMillis;
+        }
+    }
+
+    /**
+     * Gets the timeout for polling records.
+     * <p>
+     * The default value of this property is {@link #DEFAULT_POLL_TIMEOUT}.
```

Comment:
you can/should use `{@value #DEFAULT_POLL_TIMEOUT}` here in order to get the constant's value instead of a link to it.